pkg/processor/k8sprocessor/kube/informer.go

```diff
@@ -53,11 +56,27 @@ func newSharedInformer(
 	return informer
 }
 
-func informerListFuncWithSelectors(client kubernetes.Interface, namespace string, ls labels.Selector, fs fields.Selector) cache.ListFunc {
+func informerListFuncWithSelectors(logger *zap.Logger, client kubernetes.Interface, namespace string, ls labels.Selector, fs fields.Selector) cache.ListFunc {
 	return func(opts metav1.ListOptions) (runtime.Object, error) {
-		opts.LabelSelector = ls.String()
-		opts.FieldSelector = fs.String()
-		return client.CoreV1().Pods(namespace).List(context.Background(), opts)
+		podList := &api_v1.PodList{}
+		for {
```

Comment:
Should this be bounded to a timeout to prevent this from potentially running forever? I think you can create a new context, call [WithDeadline](https://pkg.go.dev/context#WithDeadline), and pass the context to the call to `List()` on line 68. This raises other questions though:
* How long should the timeout be?
* Should the timeout be configurable?

Reply:
yes, made the change to use `WithDeadline` thanks! I let the timeout be 60 seconds at the moment. I made the page size (limit) in the API call configurable.

Reply:
@rnishtala-sumo FYI you can use `context.WithTimeout(duration)` instead of `context.WithDeadline` which saves you computing the `time.Time`

Reply:
thanks!, made the change